### PR TITLE
Notification enhanced

### PR DESCRIPTION
**Pull request by lorenzoPrimi**
Fixes https://github.com/ooni/probe/issues/1242

### CODE REVIEW

File: app/src/full/java/org/openobservatory/ooniprobe/common/FirebaseMessagingService.java [hunk 1]
```diff
@@ -17,7 +21,15 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
 		// decode message data and extract meaningful information from it: title, body, badge, etc.
 		CountlyPush.Message message = CountlyPush.decodeMessage(remoteMessage.getData());
 
-		Boolean result = CountlyPush.displayMessage(getApplicationContext(), message, R.drawable.notification_icon, null);
+		Intent notificationIntent = null;
+		if (!message.has("c.l")) {
```
Comment (bassosimone) [inline app/src/full/java/org/openobservatory/ooniprobe/common/FirebaseMessagingService.java:25]: Wow, `c.l` is very cryptic! Is this data format documented in some place?

File: app/src/full/java/org/openobservatory/ooniprobe/common/FirebaseMessagingService.java [hunk 2]
```diff
@@ -17,7 +21,15 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
...
+			notificationIntent = new Intent(this, MainActivity.class);
+			notificationIntent.putExtra(MainActivity.NOTIFICATION_DIALOG, "yes");
+			notificationIntent.putExtra("title", "title");
+			notificationIntent.putExtra("message", "message");
```
Comment (bassosimone) [inline app/src/full/java/org/openobservatory/ooniprobe/common/FirebaseMessagingService.java:29]: Is is intended that here we're passing `"message"` as a string as a opposed to `message` as an object?

File: app/src/full/java/org/openobservatory/ooniprobe/common/FirebaseMessagingService.java [hunk 3]
```diff
@@ -17,7 +21,15 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
 		// decode message data and extract meaningful information from it: title, body, badge, etc.
 		CountlyPush.Message message = CountlyPush.decodeMessage(remoteMessage.getData());
 
-		Boolean result = CountlyPush.displayMessage(getApplicationContext(), message, R.drawable.notification_icon, null);
+		Intent notificationIntent = null;
+		if (!message.has("c.l")) {
+			notificationIntent = new Intent(this, MainActivity.class);
+			notificationIntent.putExtra(MainActivity.NOTIFICATION_DIALOG, "yes");
+			notificationIntent.putExtra("title", "title");
```
Comment (bassosimone) [inline app/src/full/java/org/openobservatory/ooniprobe/common/FirebaseMessagingService.java:28]: Is it intended here that the `"title"` extra attribute has the `"title"` string as value?

File: app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java [hunk 4]
```diff
@@ -109,8 +109,19 @@ else if (getPreferenceManager().getAppOpenCount() == PreferenceManager.NOTIFICAT
     @Override
     protected void onNewIntent(Intent intent) {
         super.onNewIntent(intent);
-        if (intent.getExtras() != null && intent.getExtras().containsKey(RES_ITEM))
-            bottomNavigation.setSelectedItemId(intent.getIntExtra(RES_ITEM, R.id.dashboard));
+        assert intent != null;
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java:112]: Why do we want to assert here?

**Review by bassosimone — COMMENTED** [review]
Asked a couple of questions

File: app/src/full/java/org/openobservatory/ooniprobe/common/FirebaseMessagingService.java (hunk 1 above)
Comment (lorenzoPrimi) [reply app/src/full/java/org/openobservatory/ooniprobe/common/FirebaseMessagingService.java:25]: in countly doc: https://support.count.ly/hc/en-us/articles/360037754031-Android
Search for "c.l"

File: app/src/full/java/org/openobservatory/ooniprobe/common/FirebaseMessagingService.java (hunk 2 above)
Comment (lorenzoPrimi) [reply app/src/full/java/org/openobservatory/ooniprobe/common/FirebaseMessagingService.java:29]: right, corrected.

File: app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java (hunk 4 above)
Comment (lorenzoPrimi) [reply app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java:112]: Removed, it was part of my testing code when I got a crash, intent can't be null there

**Review by bassosimone — APPROVED** [review]
🐳

**Pull request merged**